src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -466,6 +466,20 @@ case class TagFamilySizeMetric(family_size: Int,
                                var fraction: Proportion = 0,
                                var fraction_gt_or_eq_family_size: Proportion = 0) extends Metric
 
+/**
+ * Metrics produced by `GroupReadsByUmi` to describe reads passed through UMI grouping
+ * @param sam_records The number of SAM records accepted for grouping.
+ * @param filtered_non_pf The number of non-PF SAM records.
+ * @param filtered_poor_alignment The number of SAM records discarded for poor alignment.
+ * @param filtered_ns_in_umi The number of SAM records discarded due to one or more Ns in the UMI.
+ * @param filtered_umis_to_short The number of SAM records discarded due to a shorter than expected UMI.
+ */
+case class UmiGroupingMetric(sam_records: Long,
+                             filtered_non_pf: Long,
+                             filtered_poor_alignment: Long,
+                             filtered_ns_in_umi: Long,
+                             filtered_umis_to_short: Long) extends Metric
```

Comment:
I'm wondering if we should have slightly easier to understand metric names:

```suggestion
case class UmiGroupingMetric(accepted_sam_records: Long,
                             discarded_non_pf: Long,
                             discarded_poor_alignment: Long,
                             discarded_ns_in_umi: Long,
                             discarded_umis_to_short: Long) extends Metric
```